cpp/src/decisiontree/quantile/quantile.cuh

```diff
@@ -183,5 +183,63 @@ void preprocess_quantile(const T *data, const unsigned int *rowids,
   return;
 }
 
+template <typename T>
+__global__ void computeQuantilesSorted(T *quantiles, const int n_bins,
+                                       const T *sorted_data, const int length) {
+  int tid = threadIdx.x + blockIdx.x * blockDim.x;
+  double bin_width = static_cast<double>(length) / n_bins;
+  int index = int(round((tid + 1) * bin_width)) - 1;
+  // Old way of computing quantiles. Kept here for comparison.
+  // To be deleted eventually
+  // int index = (tid + 1) * floor(bin_width) - 1;
+  if (tid < n_bins) {
+    quantiles[tid] = sorted_data[index];
+  }
+
+  return;
+}
+
+template <typename T>
+void computeQuantiles(
+  T *quantiles, int n_bins, const T *data, int n_rows, int n_cols,
+  const std::shared_ptr<MLCommon::deviceAllocator> device_allocator,
```

Comment:
Do we have a test for this new method for computing quantiles?

Reply:
I have written it for my own reference while developing the code but did no add it in the PR. I think it is a good idea to add it. I'll add it soon.

Reply:
Added quantile computation test now.